--- origsrc/Thunar-0.9.91/configure.in	2008-10-14 01:51:42.000000000 -0500
+++ src/Thunar-0.9.91/configure.in	2008-11-02 11:54:18.014776300 -0600
@@ -48,6 +48,7 @@
 AC_AIX()
 AC_ISC_POSIX()
 AC_MINIX()
+AM_CONDITIONAL([HAVE_CYGWIN], [test "`uname | grep \"CYGWIN\"`" != ""])
 
 dnl ********************************
 dnl *** Check for basic programs ***
--- origsrc/Thunar-0.9.91/plugins/thunar-apr/Makefile.am	2008-06-19 03:08:11.000000000 -0500
+++ src/Thunar-0.9.91/plugins/thunar-apr/Makefile.am	2008-11-02 11:54:18.030401300 -0600
@@ -37,7 +37,13 @@
 	-module								\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+thunar_apr_la_LDFLAGS += -no-undefined
+endif
+
 thunar_apr_la_LIBADD =							\
+	$(top_builddir)/thunarx/libthunarx-1.la		\
+	$(EXO_LIBS)									\
 	$(EXIF_LIBS)
 
 EXTRA_DIST =								\
--- origsrc/Thunar-0.9.91/plugins/thunar-sbr/Makefile.am	2008-06-19 03:08:13.000000000 -0500
+++ src/Thunar-0.9.91/plugins/thunar-sbr/Makefile.am	2008-11-02 14:30:08.574343900 -0600
@@ -43,7 +43,14 @@
 	-module								\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+thunar_sbr_la_LDFLAGS += -no-undefined
+endif
+
 thunar_sbr_la_LIBADD =							\
+	$(top_builddir)/thunarx/libthunarx-1.la		\
+	$(top_builddir)/thunar-vfs/libthunar-vfs-1.la		\
+	$(EXO_LIBS)								\
 	$(PCRE_LIBS)							\
 	$(EXIF_LIBS)
 
--- origsrc/Thunar-0.9.91/plugins/thunar-uca/Makefile.am	2008-06-19 03:08:14.000000000 -0500
+++ src/Thunar-0.9.91/plugins/thunar-uca/Makefile.am	2008-11-02 11:54:18.077276300 -0600
@@ -40,6 +40,15 @@
 	-module								\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+thunar_uca_la_LDFLAGS += -no-undefined
+endif
+
+thunar_uca_la_LIBADD =									\
+	$(top_builddir)/thunar-vfs/libthunar-vfs-1.la		\
+	$(top_builddir)/thunarx/libthunarx-1.la				\
+	$(EXO_LIBS)
+
 defaultsdir = $(sysconfdir)/xdg/Thunar
 defaults_in_files = uca.xml.in
 defaults_DATA = $(defaults_in_files:.xml.in=.xml)
--- origsrc/Thunar-0.9.92/plugins/thunar-wallpaper/Makefile.am	2008-11-11 17:54:05.000000000 -0600
+++ src/Thunar-0.9.92/plugins/thunar-wallpaper/Makefile.am	2008-12-26 11:00:43.006125000 -0600
@@ -29,3 +29,12 @@
 	$(PLATFORM_LDFLAGS)						\
 	$(THUNARX_LDFLAGS)						\
 	$(THUNAR_VFS_LDFLAGS)
+
+if HAVE_CYGWIN
+thunar_wallpaper_plugin_la_LDFLAGS += -no-undefined
+endif
+
+thunar_wallpaper_plugin_la_LIBADD =		\
+	$(top_builddir)/thunar-vfs/libthunar-vfs-1.la		\
+	$(top_builddir)/thunarx/libthunarx-1.la				\
+	$(GTK_LIBS)
--- origsrc/Thunar-0.9.91/thunar-vfs/Makefile.am	2008-06-19 03:08:19.000000000 -0500
+++ src/Thunar-0.9.91/thunar-vfs/Makefile.am	2008-11-02 11:54:18.077276300 -0600
@@ -143,6 +143,10 @@
 	-version-info $(THUNAR_VERINFO)					\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+libthunar_vfs_1_la_LDFLAGS += -no-undefined
+endif
+
 libthunar_vfs_1_la_LIBADD =						\
 	$(EXO_LIBS)							\
 	$(GTHREAD_LIBS)							\
--- origsrc/Thunar-0.9.91/thunarx/Makefile.am	2008-06-19 03:07:58.000000000 -0500
+++ src/Thunar-0.9.91/thunarx/Makefile.am	2008-11-02 11:54:18.124151300 -0600
@@ -61,6 +61,10 @@
 	-version-info $(THUNAR_VERINFO)					\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+libthunarx_1_la_LDFLAGS += -no-undefined
+endif
+
 libthunarx_1_la_LIBADD =						\
 	$(GTK_LIBS)
 
