--- origsrc/Thunar-0.9.91/plugins/thunar-tpa/Makefile.am	2008-06-19 03:08:12.000000000 -0500
+++ src/Thunar-0.9.91/plugins/thunar-tpa/Makefile.am	2008-11-02 22:37:49.456316300 -0600
@@ -9,7 +9,7 @@
 	-DPACKAGE_LOCALE_DIR=\"$(localedir)\"				\
 	$(PLATFORM_CPPFLAGS)
 
-pluginsdir = $(libexecdir)/xfce4/panel-plugins
+pluginsdir = $(libdir)/xfce4/panel-plugins
 plugins_PROGRAMS =							\
 	thunar-tpa
 
