--- origsrc/Thunar-1.0.1/thunar-vfs/thunar-vfs-io-jobs.c	2009-04-04 04:03:02.000000000 -0500
+++ src/Thunar-1.0.1/thunar-vfs/thunar-vfs-io-jobs.c	2009-06-15 21:06:07.490853300 -0500
@@ -39,6 +39,10 @@
 #include <unistd.h>
 #endif
 
+#ifndef DEFFILEMODE
+#define DEFFILEMODE (S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH) /* 0666 */
+#endif
+
 #include <thunar-vfs/thunar-vfs-enum-types.h>
 #include <thunar-vfs/thunar-vfs-info.h>
 #include <thunar-vfs/thunar-vfs-io-jobs.h>
