--- origsrc/Thunar-0.8.0/configure.in	2007-01-20 14:40:37.000000000 -0600
+++ src/Thunar-0.8.0/configure.in	2007-06-03 19:25:58.781250000 -0500
@@ -48,6 +48,7 @@
 AC_AIX()
 AC_ISC_POSIX()
 AC_MINIX()
+AM_CONDITIONAL([HAVE_CYGWIN], [test "`uname | grep \"CYGWIN\"`" != ""])
 
 dnl ********************************
 dnl *** Check for basic programs ***
--- origsrc/Thunar-0.8.0/plugins/thunar-apr/Makefile.am	2007-01-20 14:39:07.000000000 -0600
+++ src/Thunar-0.8.0/plugins/thunar-apr/Makefile.am	2007-06-03 23:10:40.109375000 -0500
@@ -37,7 +37,13 @@
 	-module								\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+thunar_apr_la_LDFLAGS += -no-undefined
+endif
+
 thunar_apr_la_LIBADD =							\
+	$(top_builddir)/thunarx/libthunarx-1.la		\
+	$(EXO_LIBS)									\
 	$(EXIF_LIBS)
 
 EXTRA_DIST =								\
--- origsrc/Thunar-0.8.0/plugins/thunar-sbr/Makefile.am	2007-01-20 14:39:07.000000000 -0600
+++ src/Thunar-0.8.0/plugins/thunar-sbr/Makefile.am	2007-06-03 23:47:09.515625000 -0500
@@ -40,7 +40,13 @@
 	-module								\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+thunar_sbr_la_LDFLAGS += -no-undefined
+endif
+
 thunar_sbr_la_LIBADD =							\
+	$(top_builddir)/thunarx/libthunarx-1.la		\
+	$(EXO_LIBS)									\
 	$(PCRE_LIBS)
 
 EXTRA_DIST =								\
--- origsrc/Thunar-0.8.0/plugins/thunar-tpa/Makefile.am	2007-01-20 14:39:07.000000000 -0600
+++ src/Thunar-0.8.0/plugins/thunar-tpa/Makefile.am	2007-06-04 00:34:41.703125000 -0500
@@ -46,7 +46,7 @@
 desktop_in_in_files = thunar-tpa.desktop.in.in
 desktop_in_files = $(desktop_in_in_files:.desktop.in.in=.desktop.in)
 %.desktop.in: %.desktop.in.in
-	sed -e "s,\@libexecdir\@,$(libexecdir),g" < $< > $@
+	sed -e "s,\@libdir\@,$(libdir),g" < $< > $@
 desktop_DATA = $(desktop_in_files:.desktop.in=.desktop)
 @INTLTOOL_DESKTOP_RULE@
 
--- origsrc/Thunar-0.8.0/plugins/thunar-tpa/thunar-tpa.desktop.in.in	2007-01-20 14:39:07.000000000 -0600
+++ src/Thunar-0.8.0/plugins/thunar-tpa/thunar-tpa.desktop.in.in	2007-06-04 00:34:52.562500000 -0500
@@ -4,5 +4,5 @@
 _Name=Trash Applet
 _Comment=Display the trash can
 Icon=gnome-fs-trash-empty
-X-XFCE-Exec=@libexecdir@/xfce4/panel-plugins/thunar-tpa
+X-XFCE-Exec=@libdir@/xfce4/panel-plugins/thunar-tpa
 
--- origsrc/Thunar-0.8.0/plugins/thunar-uca/Makefile.am	2007-01-20 14:39:07.000000000 -0600
+++ src/Thunar-0.8.0/plugins/thunar-uca/Makefile.am	2007-06-04 00:11:21.093750000 -0500
@@ -40,6 +40,15 @@
 	-module								\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+thunar_uca_la_LDFLAGS += -no-undefined
+endif
+
+thunar_uca_la_LIBADD =									\
+	$(top_builddir)/thunar-vfs/libthunar-vfs-1.la		\
+	$(top_builddir)/thunarx/libthunarx-1.la				\
+	$(EXO_LIBS)
+
 defaultsdir = $(sysconfdir)/xdg/Thunar
 defaults_in_files = uca.xml.in
 defaults_DATA = $(defaults_in_files:.xml.in=.xml)
--- origsrc/Thunar-0.8.0/thunar-vfs/Makefile.am	2007-01-20 14:39:09.000000000 -0600
+++ src/Thunar-0.8.0/thunar-vfs/Makefile.am	2007-06-03 19:27:40.468750000 -0500
@@ -143,6 +143,10 @@
 	-version-info $(THUNAR_VERINFO)					\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+libthunar_vfs_1_la_LDFLAGS += -no-undefined
+endif
+
 libthunar_vfs_1_la_LIBADD =						\
 	$(EXO_LIBS)							\
 	$(GTHREAD_LIBS)							\
--- origsrc/Thunar-0.8.0/thunarx/Makefile.am	2007-01-20 14:38:58.000000000 -0600
+++ src/Thunar-0.8.0/thunarx/Makefile.am	2007-06-03 19:26:59.000000000 -0500
@@ -61,6 +61,10 @@
 	-version-info $(THUNAR_VERINFO)					\
 	$(PLATFORM_LDFLAGS)
 
+if HAVE_CYGWIN
+libthunarx_1_la_LDFLAGS += -no-undefined
+endif
+
 libthunarx_1_la_LIBADD =						\
 	$(GTK_LIBS)
 
